### PR TITLE
Introduce loading time limit and question time limit within groups

### PR DESCRIPTION
**Pull request by GeorgeChatzigiannis**
- Question and loading Timings will now have the following override sequence:
1. group timings
2. setting timings
3. default (config) timings

### CODE REVIEW

File: admin/data/sql/migrations/069.do.add-config-to-groups.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE mtc_admin.[group] ADD loadingTimeLimit tinyint NULL
```
Comment (pris54) [inline admin/data/sql/migrations/069.do.add-config-to-groups.sql]: @GeorgeChatzigiannis both fields need to be decimal(5,2)

File: admin/services/config.service.js [hunk 2]
```diff
@@ -19,8 +20,12 @@ const configService = {
     let loadingTime = TIME_BETWEEN_QUESTIONS
 
     const timeSettings = await settingDataService.sqlFindOne()
-
-    if (timeSettings) {
+    const group = await groupDataService.sqlFindOneBySchoolId(pupil.school_id)
+    const hasGroupTimeLimits = group && group.loadingTimeLimit && group.questionTimeLimit
```
Comment (GuyHarwood) [inline admin/services/config.service.js]: you cant look up the pupil group like this.  There can be many groups in a school, and the pupil may or may not be in one of them.  You should be looking up the group and its settings with a JOIN on the pupilGroup table

File: admin/data/sql/migrations/069.do.add-config-to-groups.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
...
+ALTER TABLE mtc_admin.[group] ADD questionTimeLimit tinyint NULL
```
Comment (pris54) [inline admin/data/sql/migrations/069.do.add-config-to-groups.sql]: @GeorgeChatzigiannis same as above

File: admin/services/data-access/group.data.service.js [hunk 4]
```diff
@@ -57,6 +57,28 @@ groupDataService.sqlFindOneById = async (groupId, schoolId) => {
   return R.head(result)
 }
 
+/**
+ * Get group by school id.
+ * @param schoolId
+ * @returns {Promise<void>}
+ */
+groupDataService.sqlFindOneBySchoolId = async (schoolId) => {
```
Comment (GuyHarwood) [inline admin/services/data-access/group.data.service.js]: as above, this is not the right query

File: admin/data/sql/migrations/069.do.add-time-limits-to-group.sql [hunk 5]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE mtc_admin.[group] ADD loadingTimeLimit tinyint NULL
```
Comment (pris54) [inline admin/data/sql/migrations/069.do.add-time-limits-to-group.sql]: @GeorgeChatzigiannis both fields need to be decimal(5,2)

File: admin/data/sql/migrations/069.do.add-time-limits-to-group.sql [hunk 6]
```diff
@@ -0,0 +1,2 @@
...
+ALTER TABLE mtc_admin.[group] ADD questionTimeLimit tinyint NULL
```
Comment (pris54) [inline admin/data/sql/migrations/069.do.add-time-limits-to-group.sql]: @GeorgeChatzigiannis same as above.

File: admin/data/sql/migrations/069.do.add-time-limits-to-group.sql (hunk 5 above)
Comment (GuyHarwood) [reply admin/data/sql/migrations/069.do.add-time-limits-to-group.sql]: good catch!

File: admin/services/data-access/group.data.service.js (hunk 4 above)
Comment (pris54) [inline admin/services/data-access/group.data.service.js]: @GeorgeChatzigiannis you may find more than one group with this query, so maybe `sqlFindOne...` is misleading

File: admin/services/data-access/group.data.service.js [hunk 7]
```diff
@@ -57,6 +57,28 @@ groupDataService.sqlFindOneById = async (groupId, schoolId) => {
...
+  const sql = `SELECT * 
+    FROM ${sqlService.adminSchema}.[group]
+    WHERE school_id=@schoolId`
```
Comment (pris54) [inline admin/services/data-access/group.data.service.js]: @GeorgeChatzigiannis you may want to add isDeleted=0 to make sure you're not fetching deleted groups.

**Pull request merged**